source/MCP3008support.h

```diff
@@ -122,24 +122,32 @@ class MCP3008Device
     {
         HRESULT hr = S_OK;
         
-        ULONG dataOut = FIXED_CMD_BITS;
-        ULONG dataIn = 0;
-
         // Make sure the channel number is in range.
         if (channel >= ADC_CHANNELS)
         {
             hr = DMAP_E_ADC_DOES_NOT_HAVE_REQUESTED_CHANNEL;
         }
 
+        ULONG dataIn = 0;
+        BYTE command = (BYTE)(channel & 0x03);  // Using only 3 bits of the passed in ULONG for channel ID
+
+        command |= 0x08;     // Single-Ended read - comparing the sample voltage to AGND
+        command <<= 4;      // Channel number and Single-Ended bit are in the upper 4 bits of the command byte
+
+        BYTE dataOut[4] = 
+        { 
+            0x00,       // Upper byte is not used (24 bit transfer)
+            0x01,       // Start Bit
+            command,    // 4 Command bits in top of byte
+            0x00        // MCP3008 ignores these bits.
+        };
+
```

Comment:
This approach involves more operations than the original approach, so I suspect it is slower.  It is, however easier to understand than the original code because of the comments.  Adding the comments to the original code would have resulted in fast, understandable code.  This is a minor issue, and I would not hold the pull request for it.